RedacteurPortaal.Tests/Api/ArchiveControllerTests.cs

```diff
@@ -0,0 +1,589 @@
+        {
+            Assert.AreEqual(patchArchiveRequest.MediaVideoItems[i], patchResult?.MediaVideoItems?[i]);
+        }
+        for (int i = 0; i < patchArchiveRequest.MediaAudioItems?.Count; i++)
+        {
+            Assert.AreEqual(patchArchiveRequest.MediaAudioItems[i], patchResult?.MediaAudioItems?[i]);
+        }
+        for (int i = 0; i < patchArchiveRequest.NewsItems?.Count; i++)
+        {
+            Assert.AreEqual(patchArchiveRequest.NewsItems[i], patchResult?.NewsItems?[i]);
+        }
+    }
+
+    [TestMethod]
+    public async Task CanGetAllArchives()
+    {
+        var application = new RedacteurPortaalApplication();
+        var client = application.CreateClient();
+
+        var newArchive = await client.GetFromJsonAsync<List<ArchiveDto>>("/api/Archive");
+
+        Assert.IsNotNull(newArchive);
+        Assert.IsTrue(newArchive?.Count == 0);
+    }
+
+    [TestMethod]
+    public async Task CanGetArchiveById()
+    {
+        var application = new RedacteurPortaalApplication();
+        var client = application.CreateClient();
+
+        var addArchiveRequest = ArchiveDtoBuilder.BuildSmallestArchive();
+        var archiveResult = await client.PostAsJsonAsync("/api/Archive", addArchiveRequest);
+        var resultString = await archiveResult.Content.ReadAsStringAsync();
+
+        var result = JsonSerializer.Deserialize<ArchiveDto>(resultString, new JsonSerializerOptions() { PropertyNameCaseInsensitive = true });
+
+        var newArchive = await client.GetFromJsonAsync<ArchiveDto>($"/api/Archive/{result?.Id}");
+
+        Assert.IsNotNull(newArchive);
+        Assert.IsTrue(newArchive?.Id == result?.Id);
+        Assert.IsTrue(newArchive?.Title == addArchiveRequest?.Title);
+        Assert.IsTrue(newArchive?.Label == addArchiveRequest?.Label);
+        CollectionAssert.AreEqual(addArchiveRequest?.Scripts, newArchive?.Scripts);
+    }
+
+    [TestMethod]
+    public async Task CanGetVideoItems()
+    {
+        var application = new RedacteurPortaalApplication();
+        var client = application.CreateClient();
+
+        var addArchiveRequest = ArchiveDtoBuilder.BuildSmallestArchive();
+        var archiveResult = await client.PostAsJsonAsync("/api/Archive", addArchiveRequest);
+        var resultString = await archiveResult.Content.ReadAsStringAsync();
+
+        var result = JsonSerializer.Deserialize<ArchiveDto>(resultString, new JsonSerializerOptions() { PropertyNameCaseInsensitive = true });
+        var videoItemId = Guid.NewGuid();
+        var updatedArchiveResult = await client.PostAsJsonAsync<Guid>($"/api/Archive/{result?.Id}/VideoItems", videoItemId);
+
+        var archiveWithVideos = await client.GetFromJsonAsync<ArchiveDto>($"/api/Archive/{result?.Id}");
+        Assert.AreEqual(HttpStatusCode.Created, archiveResult.StatusCode);
+        Assert.AreEqual(HttpStatusCode.OK, updatedArchiveResult.StatusCode);
+        Assert.IsNotNull(archiveWithVideos);
```

Comment:
Al deze tests kun je samenvoegen in 1, get audio, video etc in cangetbyid

Reply:
video, photo, audio en nieuws items worden nu allemaal getest met de methode "CanGetItemsById"